app/src/components/widgets/charts/investigatedTrendChart/investigatedTrendChart.jsx

```diff
@@ -136,12 +175,43 @@ export class InvestigatedTrendChart extends Component {
     } else {
       this.config = getLaunchModeConfig(params);
     }
+    this.config.data.onclick = this.onChartClick;
 
     this.setState({
       isConfigReady: true,
     });
   };
 
+  timeLineModeClickHandler = (data) => {
+    const { widget } = this.props;
+
+    const arrResult = Object.keys(widget.content.result).map((item) => item);
```

Comment:
I think you can create common util function to get new filter, f.e. `getUpdatedFilterWithTime(chartFilter, itemDate)` and reuse it here and in the `launchStatisticsChart`